core/lib/snt/core/mq/publisher.rb

```diff
@@ -46,6 +46,9 @@ def initialize
         # @return [Boolean]
         #
         def publish!(msg, options = {})
+          if OpenTracing && OpenTracing.current_trace_object
```

Comment:
This may not work. If OpenTracing isn't loaded, it will throw NameError here. This should likely be...

```
if defined?(::OpenTracing) && ::OpenTracing.current_trace_object
```